crypto/elliptic-curve.ts

```diff
@@ -530,14 +623,12 @@ function createCurveAffine({
   endoScalar,
   endoBase,
 }: CurveParams) {
-  // TODO: lift this limitation by using other formulas (in projectiveScale) for a != 0
-  if (a !== 0n) throw Error('createCurveAffine only supports a = 0');
```

Comment:
Removing this check implies you allow any `a`? In `createCurveProjective` you removed the check and called `createProjectiveDouble` where there's a check that `a` is 0 or -3. But what about here?

Reply:
that's a good point, I mean we do support general `a` now in all the purely affine methods like `affineAdd`, `affineDouble`. the only place which doesn't support general a is `projectiveDouble`, which is used here by methods like `scale()` and `isInSubgroup()`

so I think we can accept general `a` here at this point and only when using certain unsupported methods users would hit the error.